Add Node 24 to CI matrix

## Why

Node 24 is Active LTS (since Oct 2025) and is the default version most users get when they install Node today. Our CI matrix is currently Node 20 + 22 only.

During a recent upgrade, a user on **Node 24.12.0** hit this under \`claude-tempo@0.24.0\`:

\`\`\`
Error: Cannot find module './internal/observable/dom/animationFrames'
Require stack:
 - .../rxjs/dist/cjs/index.js
 - .../@temporalio/worker/lib/worker.js
\`\`\`

Root cause: Node 24 tightened CJS module resolution, and the rxjs version that \`@temporalio/worker\` transitively pulled in 0.24.0 hadn't been updated to match. **Every claude-tempo CLI command failed** under Node 24.

## What's fixed and what isn't

- \`0.25.0-beta.2\` installs and runs cleanly on Node 24 — the transitive rxjs got bumped somewhere in the v0.25 dep tree. Smoke test (recruit → cue → report → destroy) passed.
- **CI didn't catch this** because we don't run Node 24. If a future \`@temporalio/worker\` bump pulls an older rxjs again, we'd ship another broken release silently.

## Asks

1. Add Node 24 to the CI matrix in \`.github/workflows/*.yml\` (alongside 20 + 22).
2. Consider an \`overrides\` entry in \`package.json\` pinning rxjs to a Node-24-safe version, so we're defended against transitive regressions.
3. Document the supported Node version range in README.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -15,7 +15,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [18, 20, 22]
+        node-version: [18, 20, 22, 24]
 
     steps:
       - uses: actions/checkout@v6

--- a/README.md
+++ b/README.md
@@ -45,7 +45,7 @@ Each session registers as a **player** in Temporal. Players discover each other 
 npm install -g claude-tempo
 ```
 
-**Prerequisites**: [Node.js](https://nodejs.org/) 18+, [Temporal CLI](https://docs.temporal.io/cli), [Claude Code](https://claude.ai/code)
+**Prerequisites**: [Node.js](https://nodejs.org/) 20 LTS, 22 LTS, or 24 LTS (18+ minimum), [Temporal CLI](https://docs.temporal.io/cli), [Claude Code](https://claude.ai/code)
 
 ## Quick Start
 

--- a/package.json
+++ b/package.json
@@ -129,6 +129,6 @@
     "protobufjs"
   ],
   "overrides": {
-    "rxjs": ">=7.8.2"
+    "rxjs": "^7.8.1"
   }
 }

--- a/package.json
+++ b/package.json
@@ -127,5 +127,8 @@
   "trustedDependencies": [
     "@swc/core",
     "protobufjs"
-  ]
+  ],
+  "overrides": {
+    "rxjs": ">=7.8.2"
+  }
 }

